src/main/java/com/ledikom/service/BotService.java

```diff
@@ -199,6 +213,28 @@ private void executeAdminActionOnMessageReceived(final RequestFromAdmin requestF
 
         if (splitStringsFromAdminMessage.get(0).equals(AdminMessageToken.NEWS.label)) {
             sendNewsToUsers(requestFromAdmin.getPhotoPath(), splitStringsFromAdminMessage);
+        } else if (splitStringsFromAdminMessage.get(0).equals(AdminMessageToken.COUPON.label)) {
+            createNewCoupon(requestFromAdmin.getPhotoPath(), splitStringsFromAdminMessage);
+        }
+    }
+
+    private void createNewCoupon(final String photoPath, final List<String> splitStringsFromAdminMessage) {
+        ZoneId moscowZone = ZoneId.of("Europe/Moscow");
+        LocalDateTime zonedDateTime = LocalDateTime.now(moscowZone).plusDays(Long.parseLong(splitStringsFromAdminMessage.get(5)));
+        zonedDateTime = zonedDateTime.withHour(23).withMinute(59).withSecond(59);
+
+        couponService.createNewCoupon(Integer.parseInt(splitStringsFromAdminMessage.get(1)),
+                splitStringsFromAdminMessage.get(2),
+                splitStringsFromAdminMessage.get(3),
+                zonedDateTime);
+
+        List<User> usersToSendNews = userService.getAllUsersToReceiveNews();
+        String messageToUsers = BotResponses.newCoupon(splitStringsFromAdminMessage.get(4), zonedDateTime);
```

Comment:
Look how the helloCoupon button is attached to hello message and attach the button with newly created coupon here to the message, so a user can use it directly without entering /kupony menu